napari/_vispy/vendored/volume.py

```diff
@@ -959,6 +976,54 @@ def _in_loop_snippet(self):
     def _after_loop_snippet(self):
         return frag_dict[self.method]['after_loop']
 
+    @staticmethod
+    @lru_cache(maxsize=10)
```

Comment:
What's the expected/actual behaviour here if more than 10 clipping planes are added? If it doesn't work, this info should be added to a docstring somewhere

Reply:
It works just fine. This cache holds up to 10 code snippets, each one generated to allow a different amount of clipping planes. It does not limit the amount of clipping planes! I just means that if you (in the same session) clip with 1 plane, then 2, then 10, then 22, and so on up to 10 different numbers, the 11th time it will discard the first cached code snippet.

The reasoning behind it is that in normal use people will tend to increase/decrease clips by one at a time and probably stay under 10 clips in total, so caching the first ten should cover the typical use case of switching back and forth between (say) 3-4-5 planes.